WiseBuy/View Controllers/RecommendationFeedViewController.m

```diff
@@ -61,43 +61,46 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
     
     Post *post = self.posts[indexPath.row];
     
-    PFQuery *query = [PFUser query];
-    [query includeKey:@"first_name"];
-    [query includeKey:@"last_name"];
-    [query includeKey:@"image"];
-
-    [query whereKey:@"objectId" equalTo:post.user.objectId];
-    NSArray *user = query.findObjects;
-    NSLog(@"%@", user[0][@"image"]);
-    
-    // TODO: fix this later
-//    NSURL *url = [NSURL URLWithString:user[0][@"image"]];
-//    NSData *data = [NSData dataWithContentsOfURL:url];
-//    UIImage *img = [[UIImage alloc] initWithData:data];
-//
-//    cell.userImageView.image = img;
-    cell.usernameLabel.text = post.username;
-    cell.userFullNameLabel.text = [NSString stringWithFormat:@"%@%@", user[0][@"first_name"], user[0][@"last_name"]];
-    cell.itemNameLabel.text = post.itemName;
-    cell.priceLabel.text = post.price;
-    cell.sellerLabel.text = post.sellerName;
-    cell.linkLabel.text = post.itemLink;
+    dispatch_async(dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^(void){
+        PFQuery *query = [PFUser query];
+        [query includeKey:@"first_name"];
+        [query includeKey:@"last_name"];
+        [query includeKey:@"image"];
+
+        [query whereKey:@"objectId" equalTo:post.user.objectId];
+        NSArray *user = query.findObjects;
+
+        dispatch_async(dispatch_get_main_queue(), ^(void){
+            if (user) {
```

Comment:
For increased safety, let's also check that there is at least 1 object returned
if (user.count > 0)